jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -425,13 +425,12 @@ public void matchCreated(MatchCreatedEvent event) {
                     // new activations of the rule associate with a state node
                     // signal process instances of that state node
                     String ruleName = event.getMatch().getRule().getName();
-                    if ( ruleName.startsWith( "RuleFlowStateNode-" )) {
-                        int index = ruleName.indexOf( '-',
-                                                      18 );
-                        index = ruleName.indexOf( '-',
-                                                  index + 1 );
-                        String eventType = ruleName.substring( 0,
-                                                               index );
+                    String rulePrefix = "RuleFlowStateNode-";
+                    if (ruleName.startsWith(rulePrefix)) {
+                        int numberOfHyphens = (int) ruleName.chars().filter(ch -> ch == '-').count();
+                        // remove the last 2
```

Comment:
If we want to remove last 2 hyphens, why not to use just `String#lastIndexOf` method twice or in a loop? This can be much quicker in case there are many hyphens in the beginning.

And btw why do you have to remove last 2 hyphens exactly? Can you please explain? Thanks.

Reply:
will change that.

Regarding the last 2 hyphens is more complicated. The rule name is a composition of (only in case of state node, check ProcessBuilderImpl where the rule is created.

RuleFlowStateNode-Intermediate-Catch-Event-2 -3 -DROOLS_DEFAULT
RuleFlowStateNode-" + process.getId() + "-" + state.getUniqueId() + "-" + key.getNodeId() + "-" + key.getToType() 

The activatioy type is  rule + processID + node uniqueId (this is general for every rule)

Reply:
Thanks, now it looks good!